localstack/services/acm/provider.py

```diff
@@ -33,26 +53,62 @@ def describe(describe_orig, self):
                     "ValidationMethod": cert.get("ValidationMethod"),
                 }
             )
+
     for option in options:
         option["DomainName"] = domain_name = option.get("DomainName") or cert.get("DomainName")
-        option["ValidationDomain"] = option.get("ValidationDomain") or option["DomainName"]
+        validation_domain = option.get("ValidationDomain") or f"test.{domain_name.lstrip('*.')}"
+        option["ValidationDomain"] = validation_domain
         option["ValidationMethod"] = option.get("ValidationMethod") or "DNS"
         status = option.get("ValidationStatus")
-        option["ValidationStatus"] = "SUCCESS" if status in [None, "PENDING_VALIDATION"] else status
-        option["ValidationEmails"] = option.get("ValidationEmails") or [
-            "admin@%s" % self.common_name
-        ]
+        option["ValidationStatus"] = (
+            "SUCCESS" if (status is None or cert_status == "ISSUED") else status
+        )
+        if option["ValidationMethod"] == "EMAIL":
+            option["ValidationEmails"] = option.get("ValidationEmails") or [
+                f"admin@{self.common_name}"
+            ]
         test_record = {
-            "Name": "test.%s" % domain_name,
+            "Name": validation_domain,
             "Type": "CNAME",
             "Value": "test123",
         }
         option["ResourceRecord"] = option.get("ResourceRecord") or test_record
+        option["ResourceRecord"]["Name"] = option["ResourceRecord"]["Name"].replace(".*.", ".")
 
     for key, value in addenda.items():
         if not cert.get(key):
             cert[key] = value
     cert["Serial"] = str(cert.get("Serial") or "")
+
+    if cert.get("KeyAlgorithm") in ["RSA_1024", "RSA_2048"]:
+        cert["KeyAlgorithm"] = cert["KeyAlgorithm"].replace("RSA_", "RSA-")
+
+    if "InUse" not in cert:
+        cert["InUse"] = False
+
+    # add subject alternative names
+    if cert["DomainName"] not in sans:
+        sans.append(cert["DomainName"])
+    if cert["DomainName"] not in sans_summaries:
+        sans_summaries.append(cert["DomainName"])
+
+    if "HasAdditionalSubjectAlternativeNames" not in cert:
+        cert["HasAdditionalSubjectAlternativeNames"] = False
+
+    if not cert.get("ExtendedKeyUsages"):
+        cert["ExtendedKeyUsages"] = [
+            {"Name": "TLS_WEB_SERVER_AUTHENTICATION", "OID": "1.3.6.1.0.1.2.3.0"},
+            {"Name": "TLS_WEB_CLIENT_AUTHENTICATION", "OID": "1.3.6.1.0.1.2.3.4"},
+        ]
+
+    # remove attributes prior to validation
+    attrs = ["CertificateAuthorityArn", "IssuedAt", "NotAfter", "NotBefore", "Serial"]
+    if not cert.get("Status") == "ISSUED":
+        for attr in attrs:
+            cert.pop(attr, None)
+        cert["KeyUsages"] = []
+        cert["ExtendedKeyUsages"] = []
```

Comment:
Here the defaults created right above are removed again, if it is not yet `ISSUED`.
This is a bit confusing, maybe we could (in a later iteration) try to change this such that the defaults are added depending on the status (analogous to the patched function).

Reply:
Yep, good point - this is indeed a bit roundabout. Looking forward to streamlining this a bit more in the next iteration 👍